### DISCUSSION TITLE
Initialization of hp34401A over RS232 serial.

### DISCUSSION THREAD

**Discussion by Rybok** [pymeasure/pymeasure#989] [Q&A]
Hi, 

I just got my HP34401A working over serial line using a NULL modem cable (crossed (tx/rx lines).
It required a bit of investigation before I got it to work and I realize that the current implementation doesn't take care of the low level parts that are needed.

Three major things:
1. Line termination had to manually be set to a value that the hp34401A would accept. ~~("\r")~~ Corrected to: ("\n")
2.~~DTR line has to manually be enabled. No support for this in the driver so I had to set it in the serial adapter object.~~ This was initially tested using and poor cable.
3. SYST:REM command is never issued which means the HP34401A isn't configured to work in remote control mode.

I'm happy to share my implementation but for now it is done in my application file. I believe that the points stated above should be a part of the driver, hidden from the user.

What are your thoughts about this?

**Comment by BenediktBurger** [comment]
Hey @Rybok ,
thanks for asking.

If the device does not work with the current driver, the driver is broken, so please open a pull request regarding these issues.

I'm unsure about the DTR line, whether it is necessary (please check) and how to implement it best. The other points are simple and straight forward.

**Comment by LongnoseRob** [comment]
it might be helpful, if you could hare your connection setup, also would be good to know if win or linux etc..

**Comment by Rybok** [comment]
UPDATED:

Hi @LongnoseRob and @bmoneke , 

Pull request filed: https://github.com/pymeasure/pymeasure/pull/992

After investigating more I can see that in my initial post in this thread there are a few things that were incorrect. The line termination and DTR.  The patch filed this has been tested many times and it work well on my setup.
